Lib/test/test_tools/test_i18n.py

```diff
@@ -72,3 +73,20 @@ def test_POT_Creation_Date(self):
 
             # This will raise if the date format does not exactly match.
             datetime.strptime(creationDate, '%Y-%m-%d %H:%M%z')
+
+    def test_files_list(self):
+        """Make sure the directories are inspected for source files
+           bpo-31920
+        """
+        source_dir = 'pypkg'
+        text = 'Text to translate'
+        with temp_cwd(None) as cwd:
+            with temp_dir(os.path.join(cwd, source_dir)) as sdir:
```

Comment:
I think it is better to use `temp_dir(None)` and test with a path that is not a child of the current directory. If pygettext.py ignores the argument and searches from the current directory, it will pass the current test, but not the test with independent test directory.